petergreve/week2/Ass3/index.html

```diff
@@ -0,0 +1,233 @@
+		},
+		handleDownClick: function(message) {
+			this.props.moveMessage(message.messageId, true);
+		},
+		handleUpClick: function(message) {
+			this.props.moveMessage(message.messageId, false);
+		},
+		render: function () {
+
+			var style = {
+	   			color: 'blue',
+    		};
+
+			var that = this;
+			return 	<div>
+					    <ol>
+					   		{this.props.messages.map(function(message,i) {
+					     		return <div key={i}>
+					     					{message.text}
+					     					<button type="button" onClick={that.handleDeleteClick.bind(null, message)}>Delete</button>
+					     					<button type="button" onClick={that.handleArchiveClick.bind(null, message)}>Archive</button>
+					     					<button type="button" onClick={that.handleDownClick.bind(null, message)}>Move Down</button>
+					     					<button type="button" onClick={that.handleUpClick.bind(null, message)}>Move Up</button>
+					     				</div>;
+					     	})}
+					    </ol>
+					    <ol>
+					   		{this.props.archived.map(function(message,i) {
+					     		return <div style={style} key={i}>
+					     					{message.text}
+					     					<button type="button" onClick={that.handleUnarchiveClick.bind(null, message)}>Unarchive</button>
+					     				</div>;
+					     	})}
+					    </ol>
+      				</div>
+		}
+
+	});
+
+	var ErrorMessage = React.createClass({
+		render: function () {
+
+			var style = {
+	   			color: 'red',
+    		};
+			return 	<div style={style}>
+						{this.props.errorMessage}
+					</div>
+		}
+
+	});
+
+	var App = React.createClass({
+		getInitialState: function() {
+		    return {
+		    		listCount: 2,
+		    		lists: [{listId: 0, listName: "first list"},{listId: 1, listName: "second list"}],
+		    		messageCount: 4,
+		    		errorMessage: '',
+		    		allMessages: [
+		    						{messageId: 0, listId: 0, text: "testmessage for list 0", isArchived: false},
+		    						{messageId: 3, listId: 0, text: "testmessage for list 0", isArchived: true},
+		    					 	{messageId: 1, listId: 1, text: "testmessage for list 1", isArchived: false},
+									{messageId: 2 ,listId: 2, text: "testmessage for list 2", isArchived: false}
+		    					 ]
+		    	};
+		},
+		render: function() {
+			var that = this;
+			return 	<div>
+		    			<input type="text" ref={(component) => this.newListName = component} />
+						<button type="button" onClick={this.createList}>New List</button>
+						<ol>
+					   		{this.state.lists.map(function(list, i) {
+					    		return <List
+					    					key={i}
+					    					handleInput={that.handleInput}
+					    					archiveMessage={that.archiveMessage}
+					    					createMessage={that.createMessage}
+					    					deleteMessage={that.deleteMessage}
+					    					messages={this.filterMessages(list.listId)}
+					    					archived={this.filterArchived(list.listId)}
+					    					moveMessage={that.moveMessage}
+					    					list={list} />;
+					   		},this)}
+					    </ol>			
+					</div>
+		},
+		handleInput: function(message) {
```

Comment:
for the future: it's a bit annoying to read code with this kind of indentation :)